multicluster/service-mirror/cluster_watcher.go

```diff
@@ -673,22 +681,32 @@ func (rcsw *RemoteClusterServiceWatcher) Stop(cleanupState bool) {
 	rcsw.eventsQueue.ShutDown()
 }
 
-func (rcsw *RemoteClusterServiceWatcher) resolveGatewayAddress() []corev1.EndpointAddress {
+func (rcsw *RemoteClusterServiceWatcher) resolveGatewayAddress() ([]corev1.EndpointAddress, error) {
 	var gatewayEndpoints []corev1.EndpointAddress
+	var errors []error
 	for _, addr := range strings.Split(rcsw.link.GatewayAddress, ",") {
-		resolved := addr
 		ipAddr, err := net.ResolveIPAddr("ip", addr)
 		if err == nil {
-			resolved = ipAddr.String()
+			gatewayEndpoints = append(gatewayEndpoints, corev1.EndpointAddress{
+				IP: ipAddr.String(),
+			})
+		} else {
+			errors = append(errors, fmt.Errorf("Error resolving '%s': %s", addr, err))
 		}
```

Comment:
That'd be a little different because `gatewayEndpoints` would still be populated even when `net.ResolveIPAddr("ip", addr)` returned an error, and I wanted for that array to only contain valid IPs.

Reply:
Oh duh right ignore me

Reply:
in the case of an error, I wouldn't assume that ipAddr is not nil.  I think your suggestion has the risk of dereferencing a nil pointer.